### ISSUE TITLE
Hash/Checksum Attribute

### ISSUE THREAD

**Issue by jadedfire** [cloudevents/spec#848]
For things like SOX compliance and other security related matters, I strongly recommend adding an attribute that contains a hash or checksum for the data, along with an attribute (enum driven?) denoting the algorithm used to calculate it.

**Comment by clemensv** [comment]
We scoped out all in-event security features since it drives up the complexity of the spec. There are some SOAP/WS-Security veterans here who would like to avoid seeing a second ship sunk by the weight of security features. We also will want to avoid inventing anything security-related here that is specific to CloudEvents.

If we were adding something like this, I can imagine *extension specs* that map S/MIME or JOSE concepts onto CloudEvents such that the algorithms and key hints and the content-type for the encrypted payload can be expressed in CloudEvents metadata.

**Comment by duglin** [comment]
Going to propose we close this on tomorrow's call... if anyone has any concerns please speak up.

**Comment by duglin** [comment]
Scott would like to keep this open for a while and discuss

**Comment by duglin** [comment]
@clemensv @n3wscott anyone want to keep this one open? I'm going to suggest we close it. We can re-open if someone wants to pick up the work.

**Comment by duglin** [comment]
Agreed on the 9/21 call to close this issue. If someone would like to reopen it and push it forward please let us know.
